src/coreclr/jit/morph.cpp

```diff
@@ -9565,7 +9574,8 @@ GenTree* Compiler::fgMorphLeaf(GenTree* tree)
         else
 #endif
         {
-            info.compCompHnd->getFunctionFixedEntryPoint(tree->AsFptrVal()->gtFptrMethod, &addrInfo);
+            info.compCompHnd->getFunctionFixedEntryPoint(tree->AsFptrVal()->gtFptrMethod,
+                                                         prepareForUnsafeFunctionPointer, &addrInfo);
```

Comment:
Can we name this something that's not so coupled to the runtime? For example flip it to `forDelegate`?

Reply:
I originally had that. I switched it to function pointer as it seems less coupled to the runtime. But the current impl does seem heavily coupled to the Delegate concept so I can switch it.

Reply:
The part I find coupled is the `prepare`. `isRawFptr` or `isExposedFptr` also seem fine with me.